app/models/dinner.rb

```diff
@@ -1,4 +1,20 @@
 class Dinner < ActiveRecord::Base
   validates :email, email: true, case_sensitive: false
   validates :email, :dinner_name, :date, :time, :location, presence: true
+
+  before_save :generate_unique_url
+
+  def custom_url
+    "http://localhost:3000/dinners/#{self.unique_url}"
```

Comment:
Look into the [to_param](http://apidock.com/rails/ActiveRecord/Integration/ClassMethods/to_param) method, which is what the various helpers in Rails call to determine how to display an object in a URL.  In this case, you'd want

```ruby
class Dinner < ActiveRecord::Base

  def to_param
    unique_url
  end

end
```

Now things like `dinner_url(@dinner)` will use `Dinner#unique_url` as the relevant URL parameter.